patina_dxe_core/src/image.rs

```diff
@@ -73,75 +72,59 @@ extern "efiapi" fn unimplemented_entry_point(
 
 // define a stack structure for coroutine support.
 struct ImageStack {
-    stack: *const [u8],
-    len: usize,
-    allocated_pages: usize,
+    stack: Box<[u8], PageFree>,
 }
 
 impl ImageStack {
     fn new(size: usize) -> Result<Self, EfiError> {
-        let mut stack: efi::PhysicalAddress = 0;
         let len = align_up(size.max(MIN_STACK_SIZE), STACK_ALIGNMENT)?;
         // allocate an extra page for the stack guard page.
-        let allocated_pages = uefi_size_to_pages!(len) + 1;
+        let page_count = uefi_size_to_pages!(len) + 1;
 
-        // allocate the stack, newly allocated memory will have efi::MEMORY_XP already set, so we don't need to set it
-        // here
-        core_allocate_pages(efi::ALLOCATE_ANY_PAGES, efi::BOOT_SERVICES_DATA, allocated_pages, &mut stack, None)?;
+        let stack = CoreMemoryManager.allocate_pages(page_count, AllocationOptions::default())?.into_boxed_slice();
 
+        let base_address = stack.as_ptr() as efi::PhysicalAddress;
         // attempt to set the memory space attributes for the stack guard page.
         // if we fail, we should still try to continue to boot
         // the stack grows downwards, so stack here is the guard page
-        let mut attributes = match dxe_services::core_get_memory_space_descriptor(stack) {
+        let mut attributes = match dxe_services::core_get_memory_space_descriptor(base_address) {
             Ok(descriptor) => descriptor.attributes,
             Err(_) => DEFAULT_CACHE_ATTR,
         };
 
         attributes = MemoryProtectionPolicy::apply_image_stack_guard_policy(attributes);
 
-        if let Err(err) = dxe_services::core_set_memory_space_attributes(stack, UEFI_PAGE_SIZE as u64, attributes) {
+        if let Err(err) =
```

Comment:
You can use `CoreMemoryManager.set_page_attributes()` for consistency

Reply:
as discussed, offline, this is a conversation that needs @os-d involved. Created #1185 to track the conversation.